lisa/tests/base.py

```diff
@@ -868,6 +869,31 @@ class RTATestBundle(FtraceTestBundle):
       the associated task will be ignored in the noise accounting.
     """
 
+    def get_trace(self, **kwargs):
+        trace = super().get_trace(**kwargs)
+
+        prefix_regexps = []
+        for prefix in self.rtapp_profile.keys():
+            regexp = "^{}(-[0-9]+)*$".format(prefix)
+            prefix_regexps.append((prefix, re.compile(regexp)))
+
+        comms = list(itertools.chain.from_iterable(trace.get_tasks().values()))
+        self.task_map = {
```

Comment:
I'd rather not keep extra state that is not strictly needed, partly for  "ABI" compatibility of serialized objects and also because @memoized does a good job for not recomputing stuff.

As in the previous round, just inline that in the memoized `def rtapp_tasks()` so we avoid all the error handling boilerplate without loosing any safety.

Reply:
Inlining in `rtapp_tasks()` will not fix the exception when we use `bundle.trace`, which uses `get_trace()` and then build a trace view.

This patch originated just because in accessing the `trace` property we actually return a view, which requires tasks name.

Reply:
The exceptions I was referring to are the one due to potentially missing attribute, not the one about missing tasks in the trace, see:
https://github.com/douglas-raillard-arm/lisa/commit/d990e27e52346ee4b12a46dee6cf657245f899d3#diff-c8bbcb8b11af6fbc7a3012b55d342036R918